### DISCUSSION TITLE
How to make EditText box selected when enter key is pressed

### DISCUSSION THREAD

**Discussion by helpme-gamingengine** [stride3d/stride#2078] [Q&A] [answered]
I am trying to make the EditText box have focus and have a blinking caret when the enter key is pressed. I feel like this should be a basic fundamental task, but I can't seem to figure out how to make it happen programatically!

**Comment by FloccinaucinihilifilipicatesLongNames** [comment]
In theory, the (simplified) snippet below should work:
```
EditText TestEditText = MyPanel.FindVisualChildOfType<EditText>("MyEditText");

            if (Input.IsKeyPressed(Keys.Enter) && !TestEditText.IsSelectionActive)
            {
                // Enter pressed? Enable the selection of this EditText.
                TestEditText.IsSelectionActive = true;
            }
```

That said, I did seem to have an issue specifically with the enter key because it is used to exit selection when the EditText is active and interacting with UI elements seem to take several frames so in the snippet above it enters selection just to immediately exit it on the next frame.

You can try using a different key, or (if you're looking to use the Enter key regardless) a workaround I came up with for this one which is to basically introduce a general button timer in the script in Update() or any method below it.
```
float ButtonTimer;
ButtonTimer -= (float)Game.DrawTime.Elapsed.TotalSeconds; // Substracts seconds per frame.
ButtonTimer = Math.Max(0, ButtonTimer); // Makes it so that the value doesn't go lower than 0.
```

Then, enforce the selection being active by using the timer:
```
EditText TestEditText = MyPanel.FindVisualChildOfType<EditText>("MyEditText");

            if (Input.IsKeyPressed(Keys.Enter) && !TestEditText.IsSelectionActive && ButtonTimer <= 0)
            {
                TestEditText.IsSelectionActive = true;
                ButtonTimer = 1.0f; // Sets a cooldown
                // 
            }
            else if (Input.IsKeyReleased(Keys.Enter) && TestEditText.IsSelectionActive && ButtonTimer > 0)
            {
                TestEditText.IsSelectionActive = true;
                // Basically, if the enter key is released while both selection and cooldown are active, bring the selection back to active.
            }
```

I'm sure there's a better solution out there like perhaps setting the EditText as read-only to disable the Enter key behavior and then toggling read-only when the selection key is pressed but this one works for me even if a bit of a clunky patch. Let me know if this helps!

**Comment by helpme-gamingengine** [comment, marked as answer]
The reason it wasn't working for me was that I had the EditBox hidden prior to trying to make it active as well. As long as you run IsSelectionActive = true on the next frame after you mark it as visible, it works.
